### PR TITLE
LPS-185506 [CP 2.0] Change the subscription page for EWSA and OEM key types

### PR DESCRIPTION
**Pull request by pedroliveirav**
[LPS-185506](https://issues.liferay.com/browse/LPS-185506)

### CODE REVIEW

**Comment by pedroliveirav** [comment]
ci:test:sf

**Comment by pedroliveirav** [comment]
ci:test:relevant

**Comment by pedroliveirav** [comment]
ci:test:relevant

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 1]
```diff
@@ -239,6 +262,24 @@ const RequiredInformation = ({
 		navigate(urlPreviousPage, {state: {newKeyGeneratedAlert: true}});
 	};
 
+	const HasClusterNodesOption = () => {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:265]: Rename

```suggestion
	const ClusterNodesOption = () => {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 2]
```diff
@@ -485,20 +526,7 @@ const RequiredInformation = ({
...
-										minAvaliableKeysCount={
-											avaliableKeysMaximumCount -
-											usedKeysCount
-										}
-										selectedClusterNodes={
-											values.maxClusterNodes
-										}
-									/>
-								</div>
+								<HasClusterNodesOption />
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:529]:
```suggestion
								<ClusterNodesOption />
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 3]
```diff
@@ -75,12 +75,24 @@ const RequiredInformation = ({
...
+		const verificationDisabledType = () => {
+			if (infoSelectedKey.hasNotPermanentLicence) {
+				if (isOemOrEnterprise) {
+					return !values.name;
+				} else {
+					return !values.name || !values.maxClusterNodes;
+				}
+			} else {
+				return !hasFilledAtLeastOneField || hasError;
+			}
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:92]:
```suggestion
			if (infoSelectedKey.hasNotPermanentLicence) {
				if (isOemOrEnterprise) {
					return !values.name;
				} 
				
				return !values.name || !values.maxClusterNodes;
			} 
			
			return !hasFilledAtLeastOneField || hasError;	
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 4]
```diff
@@ -75,12 +75,24 @@ const RequiredInformation = ({
...
 
+	const isOemOrEnterprise =
+		infoSelectedKey?.licenseEntryType.includes('OEM') ||
+		infoSelectedKey?.licenseEntryType.includes('Enterprise');
+
 	useEffect(() => {
-		const verificationDisabledType = infoSelectedKey.hasNotPermanentLicence
-			? !values.name || !values.maxClusterNodes
-			: !hasFilledAtLeastOneField || hasError;
+		const verificationDisabledType = () => {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:83]:
```suggestion
		const getVerificationDisabledType = () => {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 5]
```diff
@@ -75,12 +75,24 @@ const RequiredInformation = ({
...
+				} else {
+					return !values.name || !values.maxClusterNodes;
+				}
+			} else {
+				return !hasFilledAtLeastOneField || hasError;
+			}
+		};
 
-		setBaseButtonDisabled(verificationDisabledType);
+		setBaseButtonDisabled(verificationDisabledType());
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:95]:
```suggestion
		setBaseButtonDisabled(getVerificationDisabledType());
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 6]
```diff
@@ -136,11 +149,21 @@ const RequiredInformation = ({
...
+			} else if (infoSelectedKey?.licenseEntryType.includes('OEM')) {
+				return 'oem';
+			} else if (
+				infoSelectedKey?.licenseEntryType.includes('Enterprise')
+			) {
+				return 'enterprise';
+			} else {
+				return 'production';
+			}
+		};
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:166]:
```suggestion
		const getLicenseEntryTypeSelected = () => {
			if (infoSelectedKey?.licenseEntryType?.includes('Virtual Cluster')) {
				return 'virtual-cluster';
			} 
			
			if (infoSelectedKey?.licenseEntryType.includes('OEM')) {
				return 'oem';
			} 
			
			if (
				infoSelectedKey?.licenseEntryType.includes('Enterprise')
			) {
				return 'enterprise';
			} 
			
			return 'production';
			
		};
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
Good job @pedroliveirav 

I left few suggestions for you

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx (hunk 1 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx]: Done. Thank you!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx (hunk 2 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx]: Done. Thank you!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx (hunk 3 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx]: Done. Thank you!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx (hunk 6 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx]: Done. Thank you!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx (hunk 4 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx]: Done. Thank you!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx (hunk 5 above)
Comment (pedroliveirav) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx]: Done. Thank you!

**Review by kevenleone — APPROVED** [review]
LGTM 🔥

**Comment by kevenleone** [comment]
ci:forward

**Comment by pedroliveirav** [comment]
ci:forward

**Comment by pedroliveirav** [comment]
ci:forward

**Comment by pedroliveirav** [comment]
ci:stop

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
Pull Request forwarded: https://github.com/brianchandotcom/liferay-portal/pull/135602

**Pull request closed without merging**